client/setec/client.go

```diff
@@ -63,38 +69,96 @@ func (s *Store) Secret(name string) []byte {
 type Client struct {
 	// Server is the URL of the secrets server to talk to.
 	Server string
+	// Roundtripper is an http.RoundTripper with which requests should
```

Comment:
Not a change request, just curious: I usually do this by taking the Client's `Do` method, on the principle that the client handles timeouts, redirects, etc., and I'm not sure if the Transport does. Is Transport the preferred interface to shim?

Reply:
Yeah Transport doesn't, we recently changed the k8s-operator from Transport to Client too because it wasn't doing retries

Reply:
TIL, I keep reaching for Transport because the cookie and redirects stuff in Client seems unnecesary, I didn't know it also did retries. Changed to use a func that matches Client.Do's signature, and documented that we default to http.DefaultClient.Do.